lesson_14/exceptions/exceptions_app/src/main/java/com/codedifferently/lesson14/ecommerce/EcommerceSystem.java

```diff
@@ -17,8 +17,20 @@ public void addProduct(String productId, String name) {
     products.put(productId, new Product(productId, name));
   }
 
-  public String placeOrder(String productId, int quantity) {
+  public String placeOrder(String productId, int quantity) throws ProductNotFoundException {
     Product product = products.get(productId);
+    /*the reason why i put the throw here is due to the fact that the product is being declared and we know in the test cases
+     * that we are given a product id, what we dont know is if thaat productId is correct so technically speaking we should check
+     * if the productId is matched up to the hashmap at all using get and if it doesnt exist then product will be null because no such key exists
+     * instead of trying to put it in every single instance we see an order id or product id our test cases check to see if the given
+     * input is even available for us to grab not if there is any input at all since that itself in my opinion would be front end and not us
+     * that would handle that sort of thing before it reaches this point
```

Comment:
Think defensively. Don't assume that your consumers are always going to do the right thing. Throwing this exception will help consumers understand why the operation didn't work in case they did the wrong thing.